### PR TITLE
pr-8 added shopping cart model:

### PR DESCRIPTION
**Pull request by nikolasdymchenko7**
- added scripts:
  - create shopping cart
  - create cart item
- added models
- added dto(s)
- added mappers
- added repositories
- added controller shopping cart

### CODE REVIEW

File: src/main/java/com/store/book/model/ShoppingCart.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Entity
+@Data
+@ToString
```
Comment (sokimaaa) [inline src/main/java/com/store/book/model/ShoppingCart.java:24]: @Data is shortcut for @ToString, @EqualsAndHashCode, @Getter on all fields, @Setter on all non-final fields, and @RequiredArgsConstructor.

So no need to add @ToString explicitly.

File: src/main/java/com/store/book/model/CartItem.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@Where(clause = "is_deleted = false")
+@NoArgsConstructor
+@Table(name = "cart_items")
+public class CartItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "shopping_cart_id", nullable = false)
+    @ToStringExclude
```
Comment (sokimaaa) [inline src/main/java/com/store/book/model/CartItem.java:30]: are you sure you want to use this annotation?

File: src/main/java/com/store/book/service/shoppingcart/impl/ShoppingCartServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        Long userId = userService.getCurrentUser().get().getId();
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        ShoppingCartDto dto = shoppingCartMapper.toDto(shoppingCart);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Transactional
+    @Override
+    public CartItemDto addCartItem(CreateCartItemRequestDto requestDto) {
+        User user = userService.getCurrentUser().get();
```
Comment (sokimaaa) [inline src/main/java/com/store/book/service/shoppingcart/impl/ShoppingCartServiceImpl.java:45]: using Optional#get without previous Optional#isPresent check is bad practice and may cause NPE. 

think about using Optional#orElse, Optional#orElseThrow

File: src/main/java/com/store/book/service/user/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -39,4 +42,14 @@ public UserResponseDto register(UserRegistrationRequestDto request)
         User savedUser = userRepository.save(user);
         return userMapper.toUserResponse(savedUser);
     }
+
+    @Override
+    public Optional<User> getCurrentUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```
Comment (sokimaaa) [inline src/main/java/com/store/book/service/user/impl/UserServiceImpl.java:48]: using Spring Context inside your business logic is a bad practice. 

it tight coupling your app and spring framework.
One of the problems of tight coupling is complex testing.

**Review by aaananas — APPROVED** [review]

**Pull request merged**